micronaut-mcp-server-java-sdk/src/main/java/io/micronaut/mcp/server/json/MicronautMcpJsonMapper.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.mcp.server.json;
+
+import io.micronaut.core.type.Argument;
+import io.micronaut.json.JsonMapper;
+import io.modelcontextprotocol.json.McpJsonMapper;
+import io.modelcontextprotocol.json.TypeRef;
+
+import java.io.IOException;
+
+/**
+ * An implementation of {@link McpJsonMapper} that uses Micronaut's {@link JsonMapper} for JSON serialization and deserialization.
+ */
+public class MicronautMcpJsonMapper implements McpJsonMapper {
+    private final JsonMapper jsonMapper;
+
+    public MicronautMcpJsonMapper(JsonMapper jsonMapper) {
+        this.jsonMapper = jsonMapper;
+    }
+
+    @Override
+    public <T> T readValue(String content, Class<T> type) throws IOException {
+        return jsonMapper.readValue(content, type);
+    }
+
+    @Override
+    public <T> T readValue(byte[] content, Class<T> type) throws IOException {
+        return jsonMapper.readValue(content, type);
+    }
+
+    @Override
+    public <T> T readValue(String content, TypeRef<T> type) throws IOException {
+        return jsonMapper.readValue(content, toArgument(type));
+    }
+
+    @Override
+    public <T> T readValue(byte[] content, TypeRef<T> type) throws IOException {
+        return jsonMapper.readValue(content, toArgument(type));
+    }
+
+    @Override
+    public <T> T convertValue(Object fromValue, Class<T> type) {
+        try {
+            byte[] bytes = jsonMapper.writeValueAsBytes(fromValue);
```

Comment:
@yawkat @dstepanov should I do `jsonMapper.readValue(fromValue.toString(), type)` instead or is this the correct way to do this?